tools/tesseratos/include/tesseratos/toolbox/plugin.hpp

```diff
@@ -0,0 +1,55 @@
+
+
+/// @dir
+/// @brief @ref toolbox-plugin plugin directory.
+
+/// @file
+/// @brief Plugin entry point.
+/// @ingroup toolbox-plugin
+
+#pragma once
+
+#include <cubos/engine/cubos.hpp>
+
+namespace tesseratos
+{
+    /// @defgroup tesseratos-toolbox-plugin Toolbox
+    /// @ingroup tesseratos
+    /// @brief Adds a resource used to keep track of whether each tool is open or not.
+    ///
+    ///
+    /// ## Resources
+    /// - @ref Toolbox
+
+    /// @brief Resource which manages other tools windows.
+    class Toolbox final
+    {
+    public:
+        std::unordered_map<std::string, bool> toolsMap;
+
+        bool isOpen(std::string toolname)
+        {
+            if (toolsMap.contains(toolname))
+            {
+                return toolsMap[toolname];
+            }
+            toolsMap[toolname] = false;
+            return false;
+        }
```

Comment:
We usually move these function definitions to the `.cpp` file. Here you would just keep the declaration
```cpp
bool isOpen(std::string toolName);
```

In the source file, you would define it:
```cpp
bool Toolbox::isOpen(std::string toolName)
{
  ...
}
```
(Same for the other methods)